pykafka/balancedconsumer.py

```diff
@@ -539,13 +556,20 @@ def consume(self, block=True):
         :param block: Whether to block while waiting for a message
         :type block: bool
         """
-
         def consumer_timed_out():
             """Indicates whether the consumer has received messages recently"""
             if self._consumer_timeout_ms == -1:
                 return False
             disp = (time.time() - self._last_message_time) * 1000.0
             return disp > self._consumer_timeout_ms
+
+        # auto-restart the internal consumer if it has stalled
+        if not self._consumer.running:
+            if not self._running:
+                return None
+            self.commit_offsets()
```

Comment:
I think this `commit_offsets` should go in `SimpleConsumer.stop()` instead, because it's basically compensating for the fact that with this pullreq, `stop` may be called unexpectedly from any worker thread. If we only do it here in `BalancedConsumer`, then running a bare instance of `SimpleConsumer` might lead to using stale offsets when an exception occurs.